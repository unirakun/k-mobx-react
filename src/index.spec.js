/* eslint-env jest */
/* eslint-disable react/jsx-filename-extension, react/prop-types, no-param-reassign */
import React from 'react'
import { Provider } from 'mobx-react'
import { types } from 'mobx-state-tree'
import { mount } from 'enzyme'
import { inject } from './index'

const Model = types
  .model({
    sub: 'model',
  })
  .named('Model')
  .actions(self => ({
    setSub: (sub) => { self.sub = sub },
  }))

const Store = types
  .model({
    some: 'data',
    array: types.array(Model),
  })
  .named('Store')

const getStore = (snapshot = { array: [{ sub: 'model 2' }, {}] }) => Store.create(snapshot)

const MyComponent = props => <div>my component {JSON.stringify(props)}</div>

describe('k-mobx-react', () => {
  const snap = (store, Wrapped, props) => {
    const wrapper = mount(<Provider store={store}><Wrapped {...props} /></Provider>)
    expect(wrapper.text()).toMatchSnapshot()
  }

  it('should works when there is no props into inject', () => {
    const Wrapped = inject(() => ({}))(MyComponent)
    snap(getStore(), Wrapped, { fromParent: true })
  })

  it('should works when there is no observables into inject', () => {
    const Wrapped = inject(() => ({ not: 'observable' }))(MyComponent)
    snap(getStore(), Wrapped, { fromParent: true })
  })

  it('should flat observables from inject', () => {
    const Wrapped = inject(({ store }) => ({ store, not: 'observable' }))(MyComponent)
    snap(getStore(), Wrapped, { fromParent: true })
  })
})
