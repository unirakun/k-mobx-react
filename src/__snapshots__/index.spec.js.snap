// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`k-mobx-react should flat observables from inject 1`] = `"my component {\\"fromParent\\":true,\\"store\\":{\\"some\\":\\"data\\",\\"array\\":[{\\"sub\\":\\"model 2\\"},{\\"sub\\":\\"model\\"}]},\\"not\\":\\"observable\\"}"`;

exports[`k-mobx-react should works when there is no observables into inject 1`] = `"my component {\\"fromParent\\":true,\\"not\\":\\"observable\\"}"`;

exports[`k-mobx-react should works when there is no props into inject 1`] = `"my component {\\"fromParent\\":true}"`;
